### ISSUE TITLE
Wrong alias suggestion: Auto add `\` at start of text

### ISSUE THREAD

**Issue by dmknght** [marlonrichert/zsh-autocomplete#208]
* `zsh-autocomplete` version: 2021-03-07
* Zsh version: debian/5.8-6
No framework or plugin manager

My .zshrc
```
# Greeting
echo "Welcome to Parrot OS"

# Prompt
PROMPT="%F{red}┌[%f%F{cyan}%m%f%F{red}]─[%f%F{yellow}%D{%H:%M-%d/%m}%f%F{red}]─[%f%F{magenta}%d%f%F{red}]%f"$'\n'"%F{red}└╼%f%F{green}$USER%f%F{yellow}$%f"
# Export PATH$
export PATH=~/.local/bin:/snap/bin:/usr/sandbox/:/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games:/usr/share/games:/usr/local/sbin:/usr/sbin:/sbin:$PATH

# alias
alias ls='ls -lh --color=auto'
alias dir='dir --color=auto'
alias vdir='vdir --color=auto'
alias grep='grep --color=auto'
alias fgrep='fgrep --color=auto'
alias egrep='egrep --color=auto'
alias python='python3'
#####################################################
# Auto completion / suggestion
# Mixing zsh-autocomplete and zsh-autosuggestions
# Requires: zsh-autocomplete (custom packaging by Parrot Team)
# Jobs: suggest files / foldername / histsory bellow the prompt
# Requires: zsh-autosuggestions (packaging by Debian Team)
# Jobs: Fish-like suggestion for command history
source /usr/share/zsh-autosuggestions/zsh-autosuggestions.zsh
source /usr/share/zsh-autocomplete/zsh-autocomplete.plugin.zsh
# Select all suggestion instead of top on result only
zstyle ':autocomplete:tab:*' insert-unambiguous yes
zstyle ':autocomplete:tab:*' widget-style menu-select
zstyle $key[Up] up-line-or-history
zstyle $key[Down] down-line-or-history
#zstyle ':autocomplete:*' max-lines 100%
#zstyle ':completion:*' list-rows-first yes
#zstyle ':completion:*' group-name ''
zstyle ':autocomplete:*' min-input 2
bindkey $key[Up] up-line-or-history
bindkey $key[Down] down-line-or-history


##################################################
# Fish like syntax highlighting
# Requires "zsh-syntax-highlighting" from apt
source /usr/share/zsh-syntax-highlighting/zsh-syntax-highlighting.zsh

# Save type history for completion and easier life
HISTFILE=~/.zsh_history
HISTSIZE=10000
SAVEHIST=10000
setopt appendhistory

alias time="/usr/bin/time -f '\t%E real,\t%U user,\t%S sys,\t%K amem,\t%M mmem'"
DEBFULLNAME="Nong Hoang Tu"
DEBEMAIL="dmknght@parrotsec.org"
export DEBFULLNAME DEBEMAIL
```
![image](https://user-images.githubusercontent.com/29118926/110707909-dffe0480-81f1-11eb-83eb-9f4ea23c794d.png)

I removed old version of plugin by `sudo apt purge  zsh-autocomplete` (i made deb package for this plugin) and remove cache `rm -rf ~/.cache/zcomp*` then reboot.
Step:
1. Type any command that has alias `ls`
2. Plugin suggests command in alias with `\` at start (last command in screenshot)
3. Press `Tab` for completion, the completion uses suggested string which contains `\`

**Comment by marlonrichert** [comment]
Fixed on the `main` branch.

**Comment by dmknght** [comment]
@marlonrichert my pc is still having this problem and so does inside VM. I deleted cache folder `~/.cache/zsh`. Do i miss any location that contains cache file?

**Comment by marlonrichert** [comment]
Try to look for any `*.zwc` files and delete those.

---

By the way, these two lines in your `.zshrc` file are nonsensical and don't do anything:
```zsh
zstyle $key[Up] up-line-or-history
zstyle $key[Down] down-line-or-history
```

**Comment by dmknght** [comment]
> Try to look for any `*.zwc` files and delete those.

I don't see anything like that in my Home folder
`By the way, these two lines in your .zshrc file are nonsensical and don't do anything:` Actually it does. When i didn't set it, the `up` and `down` shows list of history commands and it moves as circle. I set it so everything will be like `bash shell`: no history selection, just the commands show on terminal. BTW, i followed your steps in the template and this problem is still happening on my PC

**Comment by dmknght** [comment]
@marlonrichert I found the reason. The problem is `alias` was used `'`. If alias uses `"` for string, it is totally fine.
Commands
```
alias ls='ls -lh --color=auto'
alias gg="ls -l"
```
The `ls`, which has `'` is buggy
![image](https://user-images.githubusercontent.com/29118926/111007717-6b5cce80-8387-11eb-9d8f-c29adda7a156.png)

Meanwhile, the `gg` is perfectly fine
![image](https://user-images.githubusercontent.com/29118926/111007754-7b74ae00-8387-11eb-8097-edefc263fcca.png)

**Comment by marlonrichert** [comment]
@dmknght Have you updated yet to the latest commit? Because that’s not the behavior I get on the `main` branch.

**Comment by marlonrichert** [comment]
Also, that has nothing to do with the quotes. The `\` is needed to prevent the `ls` alias from re-expanding. It's not needed for the `gg` alias, because the string `gg` does not appear within its alias value.
